File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int output = 0;
        int l = 0; int r = 1;
        while (r < prices.length) {
            if (prices[l]<prices[r]) {
                output = Math.max(output, prices[r]-prices[l]);
            } else {
                l = r;
            }
            r++;
        }
        return output;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        lookup = set(nums)
        output = 0
        for num in nums:
            if num-1 not in lookup: # Starting point found
                length = 0
                while num+length in lookup:
                    length += 1
                output = max(output, length)

        return output